MapboxAndroidDemo/src/main/java/com/mapbox/mapboxandroiddemo/examples/labs/SpinningSymbolLayerIconActivity.java

```diff
@@ -0,0 +1,180 @@
+  private MapboxMap mapboxMap;
+  private ValueAnimator iconSpinningAnimator;
+
+  @Override
+  protected void onCreate(Bundle savedInstanceState) {
+    super.onCreate(savedInstanceState);
+
+    // Mapbox access token is configured here. This needs to be called either in your application
+    // object or in the same activity which contains the mapview.
+    Mapbox.getInstance(this, getString(R.string.access_token));
+
+    // This contains the MapView in XML and needs to be called after the access token is configured.
+    setContentView(R.layout.activity_spinning_symbol_layer_icon);
+
+    // Initialize the MapView
+    mapView = findViewById(R.id.mapView);
+    mapView.onCreate(savedInstanceState);
+    mapView.getMapAsync(this);
+  }
+
+  @Override
+  public void onMapReady(final MapboxMap mapboxMap) {
+    this.mapboxMap = mapboxMap;
+
+    // Create a Style object
+    mapboxMap.setStyle(new Style.Builder().fromUri(Style.MAPBOX_STREETS)
+
+      // Add the SymbolLayer icon image to the map style
+      .withImage(ICON_ID, BitmapUtils.getBitmapFromDrawable(
+        getResources().getDrawable(R.drawable.hurricane_icon)))
+
+      // Adding a GeoJson source for the SymbolLayer icons.
+      .withSource(new GeoJsonSource(SOURCE_ID, loadGeoJsonFromAsset("spinning_icon.geojson")))
+
+      // Adding the actual SymbolLayer to the map style.
+      .withLayer(new SymbolLayer(LAYER_ID, SOURCE_ID)
+        .withProperties(PropertyFactory.iconImage(ICON_ID),
+          iconAllowOverlap(true))
+      ), new Style.OnStyleLoaded() {
+        @Override
+        public void onStyleLoaded(@NonNull Style style) {
+          startIconSpinningAnimation();
+        }
+      });
+  }
+
+  /**
+   * Set up and start the icon spinning animation. The Android system ValueAnimator emits a new value, which is
+   * used as the SymbolLayer icons' rotation value.
+   */
+  private void startIconSpinningAnimation() {
+    if (iconSpinningAnimator != null) {
+      iconSpinningAnimator.cancel();
+    }
+    iconSpinningAnimator = ValueAnimator.ofFloat(0, 360);
+    iconSpinningAnimator.setDuration(DESIRED_SECONDS_PER_ONE_FULL_360_SPIN * 1000);
+    iconSpinningAnimator.setInterpolator(new LinearInterpolator());
+    iconSpinningAnimator.setRepeatCount(ValueAnimator.INFINITE);
+    iconSpinningAnimator.addUpdateListener(new ValueAnimator.AnimatorUpdateListener() {
+      @Override
+      public void onAnimationUpdate(ValueAnimator valueAnimator) {
+        // Retrieve the new animation number to use as the map camera bearing value
+        Float newIconRotateValue = (Float) valueAnimator.getAnimatedValue();
+        mapboxMap.getStyle(new Style.OnStyleLoaded() {
+          @Override
+          public void onStyleLoaded(@NonNull Style style) {
+            Layer iconSymbolLayer = style.getLayerAs(LAYER_ID);
+            iconSymbolLayer.setProperties(
+              iconRotate(newIconRotateValue),
+              iconPitchAlignment(ICON_PITCH_ALIGNMENT_MAP)
```

Comment:
`iconPitchAlignment` should be set on layer creation, it doesn't have to be repeated with each update.

Reply:
Fixed